src/Functions/ReplaceStringImpl.h

```diff
@@ -346,7 +353,21 @@ struct ReplaceStringImpl
         size_t input_rows_count)
     {
         if (needle.empty())
-            throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND, "Length of the pattern argument in function {} must be greater than 0.", name);
+        {
+            assert(input_rows_count == haystack_data.size() / n);
+            /// Since ColumnFixedString does not have a zero byte at the end, while ColumnString does,
+            /// we need to split haystack_data into strings of length n, add 1 zero byte to the end of each string
+            /// and then copy to res_data, ref: ColumnString.h and ColumnFixedString.h
```

Comment:
It would be nice to have a test with FixedString too

Reply:
Do you know how to construct a ColumnConst with string type without modifying code `useDefaultImplementationForConstants = false` ? I have tested with changing this code locally, it works expected.

Reply:
Executing of the following query
```
select replace(materialize(toFixedString('abcdef', 6)), '', '01')
```
will lead to this part of code.

Though it's strange that a more simple query
```
select replace(toFixedString('abcdef', 6), '', '01')
```
just doesn't work.